diff --git a/scripts/lib/display/Backing.js b/scripts/lib/display/Backing.js
index 8675a14..18d9f9d 100644
--- a/scripts/lib/display/Backing.js
+++ b/scripts/lib/display/Backing.js
@@ -76,16 +76,17 @@
             Backing.prototype.update = function () {
             };
 
-            Backing.prototype.draw = function (camera) {
-                var i = 0, len = this._layers.length, layer;
-                for (i; i < len; i++) {
-                    layer = this._layers[i];
-
-                    if (layer.visible) {
-                        layer.camera = camera;
-                        layer.render(this._buffer);
-                    }
-                }
+            Backing.prototype.draw = function (camera) {
+                var i = 0, len = this._layers.length, layer;
+                this._clearBuffer();
+                for (i; i < len; i++) {
+                    layer = this._layers[i];
+
+                    if (layer.visible) {
+                        layer.camera = camera;
+                        layer.render(this._buffer);
+                    }
+                }
             };
 
             Backing.prototype._clearBuffer = function () {
