diff --git a/scripts/ui/ugc/FishBowl.js b/scripts/ui/ugc/FishBowl.js
index a310c1b..d9deaed 100644
--- a/scripts/ui/ugc/FishBowl.js
+++ b/scripts/ui/ugc/FishBowl.js
@@ -171,12 +171,12 @@ var ui;
                     ui.ugcSignals.thumbnailClicked.dispatch(post);
                 }
 
-                if (ui.ugc.showFirstStamps) {
-                    var stampData = pt.asset.stampDataCache.getData('stamps');
-                    var stampImages = stampData.getTierUnlockAAImages(0);
-                    this.uiHandler.containerAction('StampUnlock', 'setMultiStampImages', stampImages);
-                    this.uiHandler.containerRequested('StampUnlock', 'Multi');
-                }
+                if (ui.ugc.showFirstStamps) {
+                    var stampData = pt.asset.stampDataCache.getData('stamps');
+                    var stampImages = stampData.getTierUnlockImages(0);
+                    this.uiHandler.containerAction('StampUnlock', 'setMultiStampImages', stampImages);
+                    this.uiHandler.containerRequested('StampUnlock', 'Multi');
+                }
 
                 lib.forceGC(lib.GCType.FULL_NO_CODE, false);
             } else {
