diff --git a/scripts/lib/sprite/PatternActor.js b/scripts/lib/sprite/PatternActor.js
index 64f15aa..6f6d449 100644
--- a/scripts/lib/sprite/PatternActor.js
+++ b/scripts/lib/sprite/PatternActor.js
@@ -158,41 +158,47 @@ var lib;
 
                 var frame = this._cells.frames[step];
 
-                ctx.fillStyle = frame.texture;
-
-                var pivotX = (this.width / 2) + this.pivot.x;
-                var pivotY = (this.height / 2) + this.pivot.y;
-
-                ctx.setTransform(1, 0, 0, 1, frame.position.x + this.position.x, frame.position.y + this.position.y);
-
-                var ctxAlpha = 0;
-                if (frame.alpha !== 1 || this.alpha !== 1) {
-                    ctxAlpha = ctx.globalAlpha;
-                    ctx.globalAlpha = frame.alpha * this.alpha;
-                }
-
-                if (frame.rotation !== 0 || this.rotation !== 0) {
-                    ctx.rotate(frame.rotation + this.rotation);
-                }
-
-                if (frame.scale.x !== 1 || frame.scale.y !== 1 || this.scale.x !== 1 || this.scale.y !== 1) {
-                    ctx.scale(frame.scale.x * this.scale.x * this._flip, frame.scale.y * this.scale.y);
-                }
-
-                ctx.translate(-pivotX, -pivotY);
-
-                var useShader = (frame.hasColor || this.saturation < 1.0);
-                if (useShader) {
-                    lib.display.ShaderManager.setRGBSBlend(frame.color[0], frame.color[1], frame.color[2], this.saturation);
-                    lib.display.ShaderManager.setRGBSPatternShader(ctx, false);
-                }
-
-                ctx.fillRect(0, 0, this.width, this.height);
-
-                if (ctxAlpha)
-                    ctx.globalAlpha = ctxAlpha;
-                if (useShader)
-                    lib.display.ShaderManager.resetPatternShader(ctx);
+                var ctxAlpha = 0;
+                var useShader = false;
+
+                ctx.save();
+                try {
+                    ctx.fillStyle = frame.texture;
+
+                    var pivotX = (this.width / 2) + this.pivot.x;
+                    var pivotY = (this.height / 2) + this.pivot.y;
+
+                    ctx.setTransform(1, 0, 0, 1, frame.position.x + this.position.x, frame.position.y + this.position.y);
+
+                    if (frame.alpha !== 1 || this.alpha !== 1) {
+                        ctxAlpha = ctx.globalAlpha;
+                        ctx.globalAlpha = frame.alpha * this.alpha;
+                    }
+
+                    if (frame.rotation !== 0 || this.rotation !== 0) {
+                        ctx.rotate(frame.rotation + this.rotation);
+                    }
+
+                    if (frame.scale.x !== 1 || frame.scale.y !== 1 || this.scale.x !== 1 || this.scale.y !== 1) {
+                        ctx.scale(frame.scale.x * this.scale.x * this._flip, frame.scale.y * this.scale.y);
+                    }
+
+                    ctx.translate(-pivotX, -pivotY);
+
+                    useShader = (frame.hasColor || this.saturation < 1.0);
+                    if (useShader) {
+                        lib.display.ShaderManager.setRGBSBlend(frame.color[0], frame.color[1], frame.color[2], this.saturation);
+                        lib.display.ShaderManager.setRGBSPatternShader(ctx, false);
+                    }
+
+                    ctx.fillRect(0, 0, this.width, this.height);
+                } finally {
+                    if (ctxAlpha)
+                        ctx.globalAlpha = ctxAlpha;
+                    if (useShader)
+                        lib.display.ShaderManager.resetPatternShader(ctx);
+                    ctx.restore();
+                }
             };
 
             PatternActor.prototype._init = function () {
