diff --git a/scripts/vsm/ViewStateManager.js b/scripts/vsm/ViewStateManager.js
index 233de4b..5f57ed5 100644
--- a/scripts/vsm/ViewStateManager.js
+++ b/scripts/vsm/ViewStateManager.js
@@ -1,988 +1,983 @@
-var vsm;
-(function (vsm) {
-    (function (LowMemThreshold) {
-        LowMemThreshold.INTERVAL = 60;
-        LowMemThreshold._timer = 0;
-
-        LowMemThreshold.WARNING_DEFAULT = 8 * 1024 * 1024;
-        LowMemThreshold.FATAL_DEFAULT = 4 * 1024 * 1024;
-        LowMemThreshold.WARNING_JSC_OBJECT = 2 * 1024 * 1024;
-
-        LowMemThreshold.MIN_ALLOC_DEFAULT = 8 * 1024 * 1024;
-    })(vsm.LowMemThreshold || (vsm.LowMemThreshold = {}));
-    var LowMemThreshold = vsm.LowMemThreshold;
-
-    vsm.playReport = null;
-    var PT1 = pt.savedata.PT1;
-
-    function reportStateDuration() {
-        if (!_currentView)
-            return;
-        var st = _currentView.enterTime;
-        var et = _currentView.exitTime > 0 ? _currentView.exitTime : nwf.system.Performance.elapsedTime;
-        switch (_viewHistory[_currentViewIndex]) {
-            case 'game':
-                vsm.playReport.addDualStat(PT1.game, (et - st) / 1000);
-                break;
-            case 'editor':
-                vsm.playReport.addDualStat(PT1.edtr, (et - st) / 1000);
-                break;
-            case 'fishbowl':
-                if (pt.ugc.mode === pt.ugc.Mode.Workshop) {
-                    vsm.playReport.addDualStat(PT1.wrkshp, (et - st) / 1000);
-                } else {
-                    vsm.playReport.addDualStat(PT1.cmnty, (et - st) / 1000);
-                }
-                break;
-            case 'level_select':
-                vsm.playReport.addDualStat(PT1.lvlslct, (et - st) / 1000);
-                break;
-        }
-    }
-    vsm.reportStateDuration = reportStateDuration;
-
-    var _dom = null;
-    var _debugMenu = null;
-    var _created = false;
-    var _maxHistoryLength = 4;
-    var _currentView = null;
-    var _soundPlayer = null;
-    var _loadingMusicPlaying = false;
-    var _musicLoadParams = null;
-
-    var _loadScreen = null;
-    var _ugcLaunchSequence = null;
-    var _uiHandler = null;
-
-    var _viewHistory = [];
-    var _currentViewIndex = -1;
-    var _nextViewIndex = -1;
-
-    var _viewData = null;
-    var _views = null;
-    var _viewInstances = {};
-
-    var _permissionsFlow = null;
-    var _miiverseSetupComplete = pt.ugc.TaskResult.None;
-
-    var _blockHBM = true;
-
-    var LoadState;
-    (function (LoadState) {
-        LoadState[LoadState["None"] = 0] = "None";
-        LoadState[LoadState["UI"] = 1] = "UI";
-        LoadState[LoadState["Init"] = 2] = "Init";
-        LoadState[LoadState["Sound"] = 3] = "Sound";
-        LoadState[LoadState["Actors_1"] = 4] = "Actors_1";
-        LoadState[LoadState["Actors_2"] = 5] = "Actors_2";
-        LoadState[LoadState["RenderData"] = 6] = "RenderData";
-        LoadState[LoadState["ViewData"] = 7] = "ViewData";
-        LoadState[LoadState["AssetDef"] = 8] = "AssetDef";
-        LoadState[LoadState["Themes"] = 9] = "Themes";
-        LoadState[LoadState["Stamps"] = 10] = "Stamps";
-        LoadState[LoadState["Music"] = 11] = "Music";
-        LoadState[LoadState["AllCaches"] = 12] = "AllCaches";
-        LoadState[LoadState["GC"] = 13] = "GC";
-        LoadState[LoadState["UGCAskPermission"] = 14] = "UGCAskPermission";
-        LoadState[LoadState["UGCExplainRestriction"] = 15] = "UGCExplainRestriction";
-
-        LoadState[LoadState["Done"] = 16] = "Done";
-    })(LoadState || (LoadState = {}));
-
-    var GC_BURN_FRAMES = 5;
-    var GC_FADE_BURN_FRAMES = 20;
-    var _loadState = LoadState.None;
-    var _loadStateQue = LoadState.None;
-    var _loadStateBurnFrames = GC_BURN_FRAMES;
-    var _stateChangeBurnFrames = GC_BURN_FRAMES;
-
-    function init() {
-        if (_created)
-            return;
-        console.log('Playtime started');
-
-        _created = true;
-        window.blockInput = true;
-
-        vsm.playReport = new pt.savedata.PlayReport();
-
-        vsm.tryConnect = true;
-        vsm.connected = nwf.net.Network.isConnected();
-        var account = nwf.act.NintendoAccountManager.getInstance().getActiveAccount();
-        vsm.miiverseBlocked = (account.miiverseRestrictionLevel == nwf.act.NintendoAccount.RESTRICTION_FULL) || !account.networkCommunicationAllowed;
-
-        _dom = lib.util.dom;
-
-        window.signals = {
-            'loadLevel': new lib.signals.Signal(),
-            'stateLoadComplete': new lib.signals.Signal(),
-            'loadTransitionComplete': new lib.signals.Signal(),
-            'showLoadScreen': new lib.signals.Signal(),
-            'advanceLoadScreen': new lib.signals.Signal(),
-            'hideLoadScreen': new lib.signals.Signal(),
-            'requestToSwitchView': new lib.signals.Signal(),
-            'netConnected': new lib.signals.Signal(),
-            'netDisconnected': new lib.signals.Signal(),
-            'showMessageBox': new lib.signals.Signal(),
-            'holdHideMessageBox': new lib.signals.Signal(),
-            'hideMessageBox': new lib.signals.Signal(),
-            'gamepadLost': new lib.signals.Signal()
-        };
-
-        window.signals.showLoadScreen.add(vsm._showLoader, vsm);
-        window.signals.advanceLoadScreen.add(vsm._advanceLoader, vsm);
-        window.signals.hideLoadScreen.add(vsm._hideLoader, vsm);
-
-        window.signals.showMessageBox.add(vsm._showMessageBox, vsm);
-        window.signals.holdHideMessageBox.add(vsm._holdHideMessageBox, vsm);
-        window.signals.hideMessageBox.add(vsm._forceHideMessageBox, vsm);
-
-        window.signals.stateLoadComplete.add(_onStateLoadComplete);
-        window.signals.requestToSwitchView.add(_switchViewTo);
-        window.signals.loadLevel.add(_setNextGameMap);
-
-        pt.input.gamePad = new lib.input.GamePad();
-        pt.input.gamePad.reset(854, 480);
-
-        pt.ugc.ugcManager = new pt.ugc.UGCManager();
-        pt.ugc.ugcHistory = new pt.ugc.UGCHistory();
-
-        pt.render.layerManager = new lib.display.LayerManager();
-
-        var bufferAttr = {
-            antialias: false,
-            stencil: false,
-            alpha: false,
-            graphicsMemory: true,
-            dirty: false
-        };
-
-        var tvBacking = new lib.display.Backing('tv', 1280, 720, bufferAttr);
-        var gpBacking = new lib.display.Backing('gamepad', 854, 480, bufferAttr);
-
-        pt.render.layerManager.addBacking(tvBacking, true, 0, 480);
-        pt.render.layerManager.addBacking(gpBacking, true, 0, 0);
-
-        pt.asset.messageStudioData = new lib.locale.MessageStudio('messageData/playtime.json');
-        var location = lib.locale.LocaleManager.getSystemLocale().split('_');
-
-        lib.layout.LayoutData.setLocale(location[0], location[1]);
-        pt.asset.messageStudioData.setLocale(location[0], location[1]);
-
-        pt.asset.initCaches();
-
-        window.signals.loadTransitionComplete.add(vsm.changeView, vsm);
-
-        pt.savedata.saveStateInit();
-        pt.savedata.saveDataInit();
-
-        _setLoadState(LoadState.UI);
-
-        _updateLoadState();
-    }
-    vsm.init = init;
-
-    function changeView(direction) {
-        if (direction && direction === ui.Fade.Out) {
-            enableHBM();
-            return;
-        }
-
-        if (_viewHistory.length > _maxHistoryLength) {
-            _viewHistory = _viewHistory.slice(-_maxHistoryLength);
-            _nextViewIndex = _viewHistory.length - 1;
-            _currentViewIndex = _nextViewIndex - 1;
-
-            console.log('[VSM.changeView] Trimming history:');
-            console.log(_viewHistory);
-        }
-
-        if (_nextViewIndex != _currentViewIndex && _nextViewIndex !== -1) {
-            _exitCurrentView();
-        } else {
-            console.log('[VSM.changeView] There is no next view');
-        }
-    }
-    vsm.changeView = changeView;
-
-    function getViewName(indexFromCurrent) {
-        if (typeof indexFromCurrent === "undefined") { indexFromCurrent = 0; }
-        return _viewHistory[_currentViewIndex + indexFromCurrent];
-    }
-    vsm.getViewName = getViewName;
-
-    function isDialogTransitioning() {
-        return _uiHandler.isTransitioning();
-    }
-    vsm.isDialogTransitioning = isDialogTransitioning;
-
-    function disableHBM() {
-        nwf.system.WiiUSystem.getInstance().homeButtonEnabled = false;
-    }
-    vsm.disableHBM = disableHBM;
-
-    function enableHBM() {
-        nwf.system.WiiUSystem.getInstance().homeButtonEnabled = !_blockHBM;
-    }
-    vsm.enableHBM = enableHBM;
-
-    function enableScreenDimming() {
-        window.nwf.display.DisplayManager.getInstance().getTVDisplay().dimmingEnabled = true;
-        window.nwf.display.DisplayManager.getInstance().getGamePadDisplay().dimmingEnabled = true;
-    }
-    vsm.enableScreenDimming = enableScreenDimming;
-
-    function disableScreenDimming() {
-        window.nwf.display.DisplayManager.getInstance().getTVDisplay().dimmingEnabled = false;
-        window.nwf.display.DisplayManager.getInstance().getGamePadDisplay().dimmingEnabled = false;
-    }
-    vsm.disableScreenDimming = disableScreenDimming;
-
-    function _showMessageBox(message, msgtype, dialogType, callback, buttonSound, dialogScreen) {
-        if (typeof msgtype === "undefined") { msgtype = ui.Dialogs.GenericDialogButton.GDB_NONE; }
-        if (typeof dialogType === "undefined") { dialogType = ui.Dialogs.GenericDialogType.GDT_ANIMATED_SMALL; }
-        if (typeof callback === "undefined") { callback = null; }
-        if (typeof buttonSound === "undefined") { buttonSound = null; }
-        if (typeof dialogScreen === "undefined") { dialogScreen = ui.Dialogs.GenericDialogScreen.GDS_BOTH; }
-        if (_uiHandler.active) {
-        }
-        _uiHandler.showMessageBox(message, msgtype, dialogType, callback, buttonSound, dialogScreen);
-
-        if (_currentView) {
-            _currentView.stage.uiHandler.block();
-        }
-    }
-    vsm._showMessageBox = _showMessageBox;
-    function _holdHideMessageBox(message) {
-        _uiHandler.holdHideMessageBox(message, _unblockOtherDialogs);
-    }
-    vsm._holdHideMessageBox = _holdHideMessageBox;
-    function _forceHideMessageBox() {
-        _uiHandler.hideMessageBox(_unblockOtherDialogs);
-    }
-    vsm._forceHideMessageBox = _forceHideMessageBox;
-
-    function _unblockOtherDialogs() {
-        if (_currentView) {
-            _currentView.stage.uiHandler.unblock();
-        }
-    }
-
-    function _exitCurrentView() {
-        if (_nextViewIndex > 0) {
-            _currentView.exit();
-
-            reportStateDuration();
-        }
-
-        var i, shared = false;
-        if (!vsm.forceClearDynamicData && _currentView && _currentView.sharedViews) {
-            i = _currentView.sharedViews.length;
-            while (i--) {
-                if (_currentView.sharedViews[i] === _viewHistory[_nextViewIndex])
-                    shared = true;
-            }
-        }
-
-        pt.render.layerManager.clearAllLayers(shared);
-
-        vsm.errMan.clearProxies();
-
-        if (!shared) {
-            pt.asset.clearDynamicCaches();
-            lib.layout.clearDynamicCaches();
-            lib.effect.clearDynamicCaches();
-            console.log('[VSM.changeView] Clear Dynamic Data');
-        }
-
-        _currentView = null;
-        _stateChangeBurnFrames = GC_BURN_FRAMES;
-
-        if (shared) {
-            lib.forceGC(lib.GCType.JS_SWEEP, true);
-        } else {
-            lib.forceGC(lib.GCType.FULL, true);
-        }
-
-        vsm.playReport.sync();
-
-        vsm.forceClearDynamicData = false;
-    }
-
-    function _enterCurrentView() {
-        console.log('[VSM.changeView] Loading ' + _viewHistory[_nextViewIndex]);
-
-        _currentViewIndex = _nextViewIndex;
-        _currentView = _viewInstances[_viewHistory[_currentViewIndex]];
-
-        pt.render.layerManager.addLayers(pt.render.renderDefLoader.getLayerConfigs(_currentView.data.layers) || []);
-
-        _currentView.enter();
-    }
-
-    function _updateBlocking() {
-        _loadScreen.draw();
-
-        if (_uiHandler.active) {
-            _uiHandler.update();
-            _uiHandler.draw();
-        }
-
-        _soundPlayer.update();
-
-        if (_loadScreen.blocking) {
-            window.webkitRequestAnimationFrame(_updateBlocking);
-        } else {
-            window.webkitRequestAnimationFrame(_update);
-        }
-    }
-
-    function _updateIdle() {
-        if (_loadScreen.active)
-            _loadScreen.draw();
-
-        if (--_stateChangeBurnFrames < 0) {
-            var allocStats = window.nwf.system.Stats.getMemoryAllocSizes();
-            if (allocStats.wkDefaultMaxAllocSize < vsm.LowMemThreshold.MIN_ALLOC_DEFAULT) {
-                lib.forceLog('[VSM] Error! LowMemThreshold.MIN_ALLOC_DEFAULT ( ' + allocStats.wkDefaultMaxAllocSize + ' )');
-                vsm.errMan.displayFatalError(nwf.system.SystemErrorCode.WEBKIT_MEM_ALLOC_FAIL);
-            }
-
-            _enterCurrentView();
-
-            window.webkitRequestAnimationFrame(_update);
-            return;
-        }
-
-        if (_uiHandler.active) {
-            _uiHandler.update();
-            _uiHandler.draw();
-        }
-
-        window.webkitRequestAnimationFrame(_updateIdle);
-    }
-
-    function _update() {
-        pt.savedata.saveStateUpdate();
-
-        if (++vsm.LowMemThreshold._timer > vsm.LowMemThreshold.INTERVAL) {
-            var memStats = nwf.system.Stats.getMemoryStats(false);
-            if (memStats.wkDefaultFreeMemory < vsm.LowMemThreshold.WARNING_DEFAULT) {
-                if (memStats.wkDefaultFreeMemory < vsm.LowMemThreshold.FATAL_DEFAULT) {
-                    lib.forceLog('[VSM] Error! LowMemThreshold.FATAL_DEFAULT ( ' + memStats.wkDefaultFreeMemory + ' )');
-                    vsm.errMan.displayFatalError(nwf.system.SystemErrorCode.WEBKIT_MEM_ALLOC_FAIL);
-                } else {
-                    lib.forceGC(lib.GCType.JS_SWEEP, false);
-                    vsm.LowMemThreshold._timer = vsm.LowMemThreshold.INTERVAL - 5;
-                }
-            } else if (memStats.jscDefaultFreeMemory < vsm.LowMemThreshold.WARNING_JSC_OBJECT) {
-                lib.forceGC(lib.GCType.JS_SWEEP, false);
-                vsm.LowMemThreshold._timer = vsm.LowMemThreshold.INTERVAL - 15;
-            } else {
-                vsm.LowMemThreshold._timer = 0;
-            }
-        }
-
-        if (!_currentView) {
-            _updateIdle();
-            return;
-        }
-
-        if (_loadScreen.blocking) {
-            _updateBlocking();
-            return;
-        }
-
-        var stage = _currentView.stage;
-
-        if (!vsm.freeze) {
-            vsm.errMan.update();
-
-            if (!window.blockInput) {
-                pt.input.gamePad.update();
-            } else {
-                pt.input.gamePad.setToIdleState();
-            }
-
-            pt.ugc.ugcManager.update();
-            if (_ugcLaunchSequence) {
-                _ugcLaunchSequence.update();
-                if (!_ugcLaunchSequence.running) {
-                    _ugcLaunchSequence.destroy();
-                    _ugcLaunchSequence = null;
-                }
-            }
-        }
-
-        pt.render.layerManager.update();
-
-        if (_currentView.active) {
-            if (!vsm.freeze) {
-                stage.update();
-            }
-
-            stage.draw();
-        }
-
-        _soundPlayer.update();
-
-        if (_loadScreen.active)
-            _loadScreen.draw();
-
-        if (_uiHandler.active) {
-            _uiHandler.update();
-            _uiHandler.draw();
-        }
-
-        window.webkitRequestAnimationFrame(_update);
-    }
-
-    function _switchViewTo(newView) {
-        if (typeof newView === "undefined") { newView = ''; }
-        var musicDef, mapAssets, themeDef;
-        var musicChannels = new Array(lib.sound.NUM_MUSIC_CHANNELS);
-        var channelIndex;
-
-        _blockHBM = true;
-        disableHBM();
-
-        vsm.errMan.setShowCommunicationErrors(false);
-        vsm.errMan.setShowBatteryWarning(false);
-
-        console.log('-------------------------------------------------------');
-
-        if (newView === '') {
-            if (_currentViewIndex === 0) {
-                console.warn('[VSM._switchViewTo] There is no view to go back to');
-                return;
-            }
-
-            newView = _viewHistory[_currentViewIndex - 1];
-            console.log('[VSM._switchViewTo] Moving to previous view: ' + newView);
-        } else if (newView === _viewHistory[_currentViewIndex]) {
-            console.warn('[VSM._switchViewTo] You are already in ' + newView + ', but continuing anyway');
-        }
-
-        _nextViewIndex = _viewHistory.length;
-        _viewHistory.push(newView);
-
-        if (_nextViewIndex > _currentViewIndex + 1) {
-            console.warn('[VSM._switchViewTo] Skipping ahead to ' + _viewHistory.pop() + '?! Stopped');
-            _nextViewIndex = _currentViewIndex;
-            return;
-        }
-
-        pt.input.gamePad.reset();
-
-        _loadScreen.setFadeDuration(ui.FadeDuration.NORMAL);
-
-        console.log('[VSM._switchViewTo] Next view at ' + _nextViewIndex + ' (currently at ' + _currentViewIndex + ')');
-        if (_currentViewIndex === -1) {
-            changeView();
-
-            _update();
-        } else {
-            switch (newView) {
-                case 'game':
-                    if (_viewHistory[_currentViewIndex] === 'cutscene') {
-                        mapAssets = pt.asset.assetDefLoader.getMapAssets();
-                        themeDef = mapAssets.getThemeTypeDef(pt.currMapDef.settings.theme);
-                        musicDef = mapAssets.getThemeMusicDef(themeDef, pt.currMapDef.settings.music);
-
-                        if (!lib.sound.SoundPlayer.instance.isMusicPlaying(musicDef.intro)) {
-                            lib.sound.SoundPlayer.instance.playMusic(musicDef.intro, true);
-                        }
-                    }
-                    if (_viewHistory[_currentViewIndex] === 'editor') {
-                        _loadScreen.setState(ui.LoadScreenState.Black, ui.ProgressType.Saving);
-                        _loadScreen.setFadeDuration(ui.FadeDuration.FAST);
-                    } else {
-                        if (pt.currGameMode === pt.GAME_MODE_SOLO || pt.currGameMode === pt.GAME_MODE_SOLO_NEXT) {
-                            _loadScreen.setState(ui.LoadScreenState.Solo, ui.ProgressType.Loading);
-                        } else {
-                            _loadScreen.setState(ui.LoadScreenState.ProTip, ui.ProgressType.Loading, pt.ugc.mode === pt.ugc.Mode.Workshop ? ui.ProTipType.Editor : ui.ProTipType.Community);
-                        }
-                    }
-
-                    break;
-                case 'editor':
-                    if (_viewHistory[_currentViewIndex] === 'game' && !vsm.forceClearDynamicData) {
-                        _loadScreen.setState(ui.LoadScreenState.Black, ui.ProgressType.Saving);
-                        _loadScreen.setFadeDuration(ui.FadeDuration.FAST);
-                    } else {
-                        _loadScreen.setState(ui.LoadScreenState.ProTip, ui.ProgressType.Loading, ui.ProTipType.Editor);
-                    }
-                    break;
-                case 'level_select':
-                    if (_viewHistory[_currentViewIndex] === 'cutscene' && pt.sceneKey !== 'credits') {
-                        for (channelIndex = 0; channelIndex < musicChannels.length; ++channelIndex) {
-                            musicChannels[channelIndex] = false;
-                        }
-
-                        musicChannels[0] = true;
-                        musicChannels[1] = true;
-
-                        lib.sound.SoundPlayer.instance.stopAllMusic(5);
-
-                        lib.sound.SoundPlayer.instance.playMusic('LEVEL_SELECT_FULL', true, 1.0, true, false, musicChannels);
-                    }
-                    _loadScreen.setState(ui.LoadScreenState.ProTip, ui.ProgressType.Loading, ui.ProTipType.Solo);
-                    break;
-                case 'bonus_select':
-                    _loadScreen.setState(ui.LoadScreenState.ProTip, ui.ProgressType.Loading, ui.ProTipType.Solo);
-                    break;
-
-                case 'fishbowl':
-                    _loadScreen.setState(ui.LoadScreenState.ProTip, ui.ProgressType.Loading, pt.ugc.mode === pt.ugc.Mode.Workshop ? ui.ProTipType.Editor : ui.ProTipType.Community);
-                    break;
-                case 'cutscene':
-                    _loadScreen.setState(ui.LoadScreenState.Black, ui.ProgressType.None);
-                    _loadScreen.setFadeDuration(ui.FadeDuration.FAST);
-                    break;
-
-                default:
-                    _loadScreen.setState(ui.LoadScreenState.Normal, ui.ProgressType.Loading);
-                    break;
-            }
-
-            _loadScreen.fadeIn();
-        }
-    }
-
-    function _setLoadState(loadState) {
-        if (_loadState !== LoadState.GC) {
-            _loadStateQue = loadState;
-            _loadState = LoadState.GC;
-        } else {
-            _loadState = loadState;
-        }
-
-        switch (_loadState) {
-            case LoadState.UI:
-                lib.display.ShaderManager.loadShaders(lib.util.CanvasUtil.ctx);
-
-                _loadScreen = new ui.LoadingScreen();
-
-                _uiHandler = new ui.GenericDialogHandler();
-
-                pt.asset.preloadAllLayoutData();
-
-                break;
-            case LoadState.Init:
-                pt.storage.init();
-
-                vsm.playReport.init();
-
-                break;
-            case LoadState.Sound:
-                _soundPlayer = lib.sound.SoundPlayer.instance = new lib.sound.SoundPlayer('audio/json/', 'audio/sounds/');
-
-                break;
-            case LoadState.Actors_1:
-                pt.asset.preloadAllEntityActorData(pt.asset.ActorLoadPhase.First);
-
-                break;
-            case LoadState.Actors_2:
-                pt.asset.preloadAllEntityActorData(pt.asset.ActorLoadPhase.Last);
-
-                break;
-            case LoadState.RenderData:
-                pt.render.renderDefLoader = new pt.render.RenderDefLoader();
-                pt.render.renderDefLoader.load('json/layerdef.json');
-
-                break;
-            case LoadState.ViewData:
-                lib.curl(['json!json/views.json'], _onViewDataLoaded);
-
-                break;
-            case LoadState.AssetDef:
-                pt.asset.assetDefLoader = new pt.map.AssetDefLoader();
-                pt.asset.assetDefLoader.load('json/map/entitydef.json', 'json/map/themedef.json', 'json/map/tiledef.json', 'json/map/musicdef.json', 'json/map/PlaytimeLevels.json');
-
-                break;
-            case LoadState.Themes:
-                pt.asset.preloadAllThemeData();
-
-                break;
-            case LoadState.Stamps:
-                pt.asset.preloadAllStampData();
-
-                break;
-            case LoadState.Music:
-                _musicLoadParams = null;
-
-                break;
-            case LoadState.AllCaches:
-                break;
-            case LoadState.UGCAskPermission:
-                _permissionsFlow = new pt.ugc.UGCPermissionFlow();
-                _permissionsFlow.ugcAskPermission();
-                _loadScreen['_hideSysAnims']();
-                window.blockInput = false;
-                vsm.errMan.setShowCommunicationErrors(true);
-                break;
-            case LoadState.UGCExplainRestriction:
-                _loadScreen['_hideSysAnims']();
-                window.blockInput = false;
-                _explainRestriction();
-
-                break;
-            case LoadState.Done:
-                vsm.playReport.addLifetimeStat(PT1.runs, 1);
-                vsm.playReport.setSessionStat(PT1.region, nwf.system.WiiUSystem.getInstance().regionCode);
-                vsm.playReport.setSessionStat(PT1.lang, nwf.system.WiiUSystem.getInstance().languageCode);
-                vsm.playReport.setSessionStat(PT1.cnt, nwf.system.WiiUSystem.getInstance().countryCode);
-                vsm.playReport.setSessionStat(PT1.reload, nwf.system.WiiUSystem.getInstance().reloadCount);
-                vsm.playReport.addLifetimeStat(PT1.reload, nwf.system.WiiUSystem.getInstance().reloadCount > 0 ? 1 : 0);
-
-                if (!vsm.connected)
-                    vsm.playReport.addLifetimeStat(PT1.offline_net, 1);
-                if (vsm.miiverseBlocked)
-                    vsm.playReport.addLifetimeStat(PT1.offline_pc, 1);
-
-                if (pt.ugc.ugcManager.isLaunchedFromMiiverse() && pt.storage.getUGCCurrent().permissionGranted && nwf.act.NintendoAccountManager.getInstance().getActiveAccount().networkCommunicationAllowed) {
-                    pt.ugc.mode = pt.ugc.Mode.Community;
-                    ui.ugc.fromTitle = false;
-                    ui.ugc.fromBoot = true;
-                    _switchViewTo('fishbowl');
-                } else {
-                    _switchViewTo(_viewData.startView);
-                }
-
-                break;
-            case LoadState.GC:
-                if (_loadStateQue === LoadState.Init) {
-                    _loadScreen.setState(ui.LoadScreenState.ProTip, ui.ProgressType.FourSteps, ui.ProTipType.Solo);
-                    _loadScreen.fadeIn();
-                    _loadStateBurnFrames = GC_FADE_BURN_FRAMES;
-                } else if (_loadStateQue === LoadState.Actors_2) {
-                    window.signals.advanceLoadScreen.dispatch();
-                    _loadScreen.changeTip(ui.ProTipType.Community);
-                    _loadStateBurnFrames = GC_FADE_BURN_FRAMES;
-
-                    lib.util.dom.removeElement('#gp-container');
-                    lib.util.dom.removeElement('#tv-container');
-                    lib.util.dom.removeElement('#loadscreen-style');
-                } else if (_loadStateQue === LoadState.RenderData) {
-                    if (!pt.ugc.ugcManager.isLaunchedFromMiiverse()) {
-                        window.signals.advanceLoadScreen.dispatch();
-                        _loadScreen.changeTip(ui.ProTipType.Editor);
-                    }
-                    _loadStateBurnFrames = GC_FADE_BURN_FRAMES;
-                } else if (_loadStateQue === LoadState.AllCaches) {
-                    window.signals.advanceLoadScreen.dispatch();
-                    if (pt.ugc.ugcManager.isLaunchedFromMiiverse()) {
-                        _loadScreen.changeTip(ui.ProTipType.Editor);
-                    }
-                    _loadStateBurnFrames = GC_FADE_BURN_FRAMES;
-                } else {
-                    _loadStateBurnFrames = GC_BURN_FRAMES;
-                }
-
-                if (_loadStateQue === LoadState.AllCaches) {
-                    lib.forceGC(lib.GCType.FULL, false);
-                } else {
-                    lib.forceGC(lib.GCType.FULL_NO_CODE, false);
-                }
-
-                break;
-        }
-
-        console.log('[VSM] Set Load State >> ' + LoadState[_loadState]);
-    }
-
-    function _updateLoadState() {
-        switch (_loadState) {
-            case LoadState.UI:
-                if (_loadScreen.ready && lib.layout.layoutDataSetCache.isReady() && _uiHandler.isReady()) {
-                    _setLoadState(LoadState.Init);
-                }
-
-                break;
-            case LoadState.Init:
-                if (pt.storage.isReady) {
-                    if (pt.storage.isCorrupt && vsm.errMan) {
-                        vsm.errMan.displayAlert(pt.asset.messageStudioData.getText('savememory_main_corrupt').text, pt.asset.messageStudioData.getText('btn_OK_01').text);
-                    }
-
-                    var ugcCurrent = pt.storage.getUGCCurrent();
-                    var activeAccount = nwf.act.NintendoAccountManager.getInstance().getActiveAccount();
-
-                    var postCache = pt.ugc.ugcManager.getPostCache(pt.ugc.PostCacheType.Shared);
-                    for (var levelIndex = 0; levelIndex < pt.ugc.MAX_MYLEVEL_COUNT; ++levelIndex) {
-                        var workshopStatus = pt.storage.getWorkshopStatus(levelIndex);
-                        if (workshopStatus.shared) {
-                            pt.ugc.ugcHistory.addLevelReference(workshopStatus.postID);
-                            pt.ugc.ugcHistory.setLevelYeahCount(workshopStatus.postID, workshopStatus.yeahs);
-                            pt.ugc.ugcHistory.setLevelReplyCount(workshopStatus.postID, workshopStatus.replyCount);
-
-                            var postItem = postCache.getItem(levelIndex);
-                            postItem.setSavedPostID(workshopStatus.postID);
-                            postItem.setSavedPosterID(activeAccount.principalID);
-                        }
-                    }
-
-                    postCache = pt.ugc.ugcManager.getPostCache(pt.ugc.PostCacheType.Saved);
-                    for (var levelIndex = 0; levelIndex < pt.ugc.MAX_SAVE_COUNT; ++levelIndex) {
-                        var ugcProgress = pt.storage.getUGCProgress(levelIndex);
-                        if (ugcProgress.postID) {
-                            pt.ugc.ugcHistory.addLevelReference(ugcProgress.postID);
-                            pt.ugc.ugcHistory.markLevelYeahed(ugcProgress.postID, ugcProgress.yeahed);
-                            pt.ugc.ugcHistory.setLevelYeahCount(ugcProgress.postID, ugcProgress.yeahs);
-                            pt.ugc.ugcHistory.markLevelPlayed(ugcProgress.postID, ugcProgress.played);
-                            pt.ugc.ugcHistory.setLevelTrophy(ugcProgress.postID, ugcProgress.trophy);
-                            pt.ugc.ugcHistory.setLevelReplyCount(ugcProgress.postID, ugcProgress.replyCount);
-
-                            var postItem = postCache.getItem(levelIndex);
-                            postItem.setSavedPostID(ugcProgress.postID);
-                            postItem.setSavedPosterID(ugcProgress.posterID);
-                        }
-                    }
-
-                    if (vsm.connected) {
-                        if (ugcCurrent.permissionGranted) {
-                            _ugcLaunchSequence = new pt.ugc.LaunchSequence();
-                            _ugcLaunchSequence.run();
-                        } else {
-                            if (pt.ugc.ugcManager.isLaunchedFromMiiverse()) {
-                                pt.ugc.ugcManager.trySetupMiiverse();
-                            }
-                        }
-                    }
-
-                    _setLoadState(LoadState.Sound);
-                }
-
-                break;
-            case LoadState.Sound:
-                if (_soundPlayer.areJSONLoaded && !_loadingMusicPlaying) {
-                    _loadingMusicPlaying = true;
-                    _soundPlayer.playMusic('LOADING', true, 1.0, true, true);
-                } else if (_soundPlayer.isLoaded()) {
-                    _setLoadState(LoadState.Actors_1);
-                }
-
-                break;
-            case LoadState.Actors_1:
-                if (pt.asset.actorDataCache.isReady() && pt.asset.patternActorDataCache.isReady()) {
-                    _setLoadState(LoadState.Actors_2);
-                }
-                break;
-            case LoadState.Actors_2:
-                if (pt.asset.actorDataCache.isReady() && pt.asset.patternActorDataCache.isReady()) {
-                    _setLoadState(LoadState.RenderData);
-                }
-
-                break;
-            case LoadState.RenderData:
-                if (pt.render.renderDefLoader.areRenderDefFilesLoaded()) {
-                    _setLoadState(LoadState.ViewData);
-                }
-
-                break;
-            case LoadState.ViewData:
-                break;
-            case LoadState.AssetDef:
-                if (pt.asset.assetDefLoader.areAssetDefFilesLoaded()) {
-                    _setLoadState(LoadState.Themes);
-                }
-
-                break;
-            case LoadState.Themes:
-                if (pt.asset.themeCache.isReady()) {
-                    _setLoadState(LoadState.Stamps);
-                }
-
-                break;
-            case LoadState.Stamps:
-                if (pt.asset.stampDataCache.isReady()) {
-                    _setLoadState(LoadState.Music);
-                }
-
-                break;
-            case LoadState.Music:
-                if (_musicLoadParams === null || _musicLoadParams.isLoaded) {
-                    _setLoadState(LoadState.AllCaches);
-                }
-
-                break;
-            case LoadState.AllCaches:
-                if (pt.asset.areCachesReady()) {
-                    if (pt.ugc.ugcManager.isLaunchedFromMiiverse()) {
-                        if (!pt.storage.getUGCCurrent().permissionGranted) {
-                            _setLoadState(LoadState.UGCAskPermission);
-                        } else if (!nwf.act.NintendoAccountManager.getInstance().getActiveAccount().networkCommunicationAllowed) {
-                            _setLoadState(LoadState.UGCExplainRestriction);
-                        } else {
-                            _setLoadState(LoadState.Done);
-                        }
-                    } else {
-                        _setLoadState(LoadState.Done);
-                    }
-                }
-
-                break;
-            case LoadState.UGCAskPermission:
-                if (_permissionsFlow) {
-                    vsm.errMan.update();
-                    _permissionsFlow.update();
-
-                    if (_permissionsFlow.permissionsAccepted || _permissionsFlow.permissionsCanceled) {
-                        window.blockInput = true;
-                        vsm.errMan.setShowCommunicationErrors(false);
-                        if (_permissionsFlow.permissionsCanceled)
-                            window.signals.advanceLoadScreen.dispatch();
-                        _loadScreen['_showSysAnims']();
-                        _permissionsFlow.destroy();
-                        _permissionsFlow = null;
-                        _setLoadState(LoadState.Done);
-                    }
-                }
-
-                break;
-            case LoadState.GC:
-                if (--_loadStateBurnFrames < 0) {
-                    _setLoadState(_loadStateQue || 0);
-                }
-
-                break;
-        }
-
-        if (_loadScreen && _loadScreen.active) {
-            _loadScreen.draw();
-        }
-
-        if (_uiHandler && _uiHandler.active) {
-            if (!window.blockInput) {
-                pt.input.gamePad.update();
-            } else {
-                pt.input.gamePad.setToIdleState();
-            }
-            _uiHandler.update();
-            _uiHandler.draw();
-        }
-
-        pt.ugc.ugcManager.update();
-        if (_ugcLaunchSequence) {
-            _ugcLaunchSequence.update();
-            if (!_ugcLaunchSequence.running) {
-                _ugcLaunchSequence.destroy();
-                _ugcLaunchSequence = null;
-            }
-        }
-
-        if (_soundPlayer) {
-            _soundPlayer.update();
-        }
-
-        if (_loadState != LoadState.Done) {
-            window.webkitRequestAnimationFrame(_updateLoadState);
-        }
-    }
-
-    function _onViewDataLoaded(localViewData) {
-        if (!localViewData)
-            throw new Error('[VSM._onViewDataLoaded] views.json did not load!');
-
-        _viewData = localViewData;
-        _views = _viewData.views;
-
-        var viewArray = [];
-        var namespaceName;
-        var className;
-
-        for (var v in _views) {
-            namespaceName = _views[v].run['namespace'];
-            className = _views[v].run['class'];
-            if (window[namespaceName] && window[namespaceName][className]) {
-                _viewInstances[v] = new vsm.View(v, _views[v]);
-                viewArray.push(v);
-            }
-        }
-
-        viewArray = null;
-
-        _setLoadState(LoadState.AssetDef);
-    }
-
-    function _onStateLoadComplete() {
-        if (_loadScreen.active) {
-            console.log('[VSM._onStateLoadComplete] Fading out load screen...');
-            _loadScreen.fadeOut();
-        } else {
-            window.blockInput = false;
-        }
-
-        if (!vsm.forceHBMDisabled) {
-            _blockHBM = false;
-        }
-        vsm.forceHBMDisabled = false;
-
-        if (_currentView) {
-            if (_currentView.data.communication) {
-                vsm.errMan.setShowCommunicationErrors(true);
-            }
-
-            vsm.errMan.setShowBatteryWarning(true);
-        }
-    }
-
-    function _setNextGameMap(mapDef, gameMode, worldIndex, levelIndex) {
-        pt.currMapDef = mapDef;
-        pt.currWorldIndex = worldIndex;
-        pt.currLevelIndex = levelIndex;
-        pt.currGameMode = gameMode;
-        console.log('[VSM] Setting GameMode to ' + gameMode);
-
-        if (gameMode === pt.GAME_MODE_SOLO) {
-            if (pt.isCutsceneNeeded(worldIndex, levelIndex)) {
-                pt.sceneExitTo = 'game';
-            } else {
-                window.signals.requestToSwitchView.dispatch('game');
-                return;
-            }
-            window.signals.requestToSwitchView.dispatch('cutscene');
-        } else {
-            window.signals.requestToSwitchView.dispatch('game');
-        }
-    }
-    vsm._setNextGameMap = _setNextGameMap;
-
-    function _showLoader(state, progressType, proTipType, fadeDuration) {
-        if (typeof state === "undefined") { state = ui.LoadScreenState.Normal; }
-        if (typeof progressType === "undefined") { progressType = ui.ProgressType.Loading; }
-        if (typeof proTipType === "undefined") { proTipType = ui.ProTipType.None; }
-        if (typeof fadeDuration === "undefined") { fadeDuration = ui.FadeDuration.NORMAL; }
-        _loadScreen.setState(state, progressType, proTipType);
-        _loadScreen.setFadeDuration(fadeDuration);
-        _loadScreen.fadeIn();
-    }
-    vsm._showLoader = _showLoader;
-
-    function _advanceLoader() {
-        _loadScreen.goToNextMarker();
-    }
-    vsm._advanceLoader = _advanceLoader;
-
-    function _hideLoader() {
-        _loadScreen.fadeOut();
-    }
-    vsm._hideLoader = _hideLoader;
-
-    function _onSetupMiiverse(taskResult) {
-        _miiverseSetupComplete = taskResult;
-        pt.ugc.ugcManager.removeListener(pt.ugc.TaskType.SetupMiiverse, _onSetupMiiverse);
-    }
-
-    function _explainRestriction() {
-        var msg = pt.asset.messageStudioData.getText('restrict_online_interaction');
-        window.signals.showMessageBox.dispatch(msg, ui.Dialogs.GenericDialogButton.GDB_OK, ui.Dialogs.GenericDialogType.GDT_FULL, _onExplainRestrictionExit);
-    }
-
-    function _onExplainRestrictionExit() {
-        window.blockInput = true;
-        _loadScreen['_showSysAnims']();
-        _setLoadState(LoadState.Done);
-    }
-})(vsm || (vsm = {}));
-//@ sourceMappingURL=ViewStateManager.js.map
+var vsm;
+(function (vsm) {
+    (function (LowMemThreshold) {
+        LowMemThreshold.INTERVAL = 60;
+        LowMemThreshold._timer = 0;
+
+        LowMemThreshold.WARNING_DEFAULT = 8 * 1024 * 1024;
+        LowMemThreshold.FATAL_DEFAULT = 4 * 1024 * 1024;
+        LowMemThreshold.WARNING_JSC_OBJECT = 2 * 1024 * 1024;
+
+        LowMemThreshold.MIN_ALLOC_DEFAULT = 8 * 1024 * 1024;
+    })(vsm.LowMemThreshold || (vsm.LowMemThreshold = {}));
+    var LowMemThreshold = vsm.LowMemThreshold;
+
+    vsm.playReport = null;
+    var PT1 = pt.savedata.PT1;
+
+    function reportStateDuration() {
+        if (!_currentView)
+            return;
+        var st = _currentView.enterTime;
+        var et = _currentView.exitTime > 0 ? _currentView.exitTime : nwf.system.Performance.elapsedTime;
+        switch (_viewHistory[_currentViewIndex]) {
+            case 'game':
+                vsm.playReport.addDualStat(PT1.game, (et - st) / 1000);
+                break;
+            case 'editor':
+                vsm.playReport.addDualStat(PT1.edtr, (et - st) / 1000);
+                break;
+            case 'fishbowl':
+                if (pt.ugc.mode === pt.ugc.Mode.Workshop) {
+                    vsm.playReport.addDualStat(PT1.wrkshp, (et - st) / 1000);
+                } else {
+                    vsm.playReport.addDualStat(PT1.cmnty, (et - st) / 1000);
+                }
+                break;
+            case 'level_select':
+                vsm.playReport.addDualStat(PT1.lvlslct, (et - st) / 1000);
+                break;
+        }
+    }
+    vsm.reportStateDuration = reportStateDuration;
+
+    var _dom = null;
+    var _debugMenu = null;
+    var _created = false;
+    var _maxHistoryLength = 4;
+    var _currentView = null;
+    var _soundPlayer = null;
+    var _loadingMusicPlaying = false;
+    var _musicLoadParams = null;
+
+    var _loadScreen = null;
+    var _ugcLaunchSequence = null;
+    var _uiHandler = null;
+
+    var _viewHistory = [];
+    var _currentViewIndex = -1;
+    var _nextViewIndex = -1;
+
+    var _viewData = null;
+    var _views = null;
+    var _viewInstances = {};
+    var _chromiumPortFrameCounter = 0;
+
+    var _permissionsFlow = null;
+    var _miiverseSetupComplete = pt.ugc.TaskResult.None;
+
+    var _blockHBM = true;
+
+    var LoadState;
+    (function (LoadState) {
+        LoadState[LoadState["None"] = 0] = "None";
+        LoadState[LoadState["UI"] = 1] = "UI";
+        LoadState[LoadState["Init"] = 2] = "Init";
+        LoadState[LoadState["Sound"] = 3] = "Sound";
+        LoadState[LoadState["Actors_1"] = 4] = "Actors_1";
+        LoadState[LoadState["Actors_2"] = 5] = "Actors_2";
+        LoadState[LoadState["RenderData"] = 6] = "RenderData";
+        LoadState[LoadState["ViewData"] = 7] = "ViewData";
+        LoadState[LoadState["AssetDef"] = 8] = "AssetDef";
+        LoadState[LoadState["Themes"] = 9] = "Themes";
+        LoadState[LoadState["Music"] = 10] = "Music";
+        LoadState[LoadState["AllCaches"] = 11] = "AllCaches";
+        LoadState[LoadState["GC"] = 12] = "GC";
+        LoadState[LoadState["UGCAskPermission"] = 13] = "UGCAskPermission";
+        LoadState[LoadState["UGCExplainRestriction"] = 14] = "UGCExplainRestriction";
+
+        LoadState[LoadState["Done"] = 15] = "Done";
+    })(LoadState || (LoadState = {}));
+
+    var GC_BURN_FRAMES = 5;
+    var GC_FADE_BURN_FRAMES = 20;
+    var _loadState = LoadState.None;
+    var _loadStateQue = LoadState.None;
+    var _loadStateBurnFrames = GC_BURN_FRAMES;
+    var _stateChangeBurnFrames = GC_BURN_FRAMES;
+
+    function init() {
+        if (_created)
+            return;
+        console.log('Playtime started');
+
+        _created = true;
+        window.blockInput = true;
+
+        vsm.playReport = new pt.savedata.PlayReport();
+
+        vsm.tryConnect = true;
+        vsm.connected = nwf.net.Network.isConnected();
+        var account = nwf.act.NintendoAccountManager.getInstance().getActiveAccount();
+        vsm.miiverseBlocked = (account.miiverseRestrictionLevel == nwf.act.NintendoAccount.RESTRICTION_FULL) || !account.networkCommunicationAllowed;
+
+        _dom = lib.util.dom;
+
+        window.signals = {
+            'loadLevel': new lib.signals.Signal(),
+            'stateLoadComplete': new lib.signals.Signal(),
+            'loadTransitionComplete': new lib.signals.Signal(),
+            'showLoadScreen': new lib.signals.Signal(),
+            'advanceLoadScreen': new lib.signals.Signal(),
+            'hideLoadScreen': new lib.signals.Signal(),
+            'requestToSwitchView': new lib.signals.Signal(),
+            'netConnected': new lib.signals.Signal(),
+            'netDisconnected': new lib.signals.Signal(),
+            'showMessageBox': new lib.signals.Signal(),
+            'holdHideMessageBox': new lib.signals.Signal(),
+            'hideMessageBox': new lib.signals.Signal(),
+            'gamepadLost': new lib.signals.Signal()
+        };
+
+        window.signals.showLoadScreen.add(vsm._showLoader, vsm);
+        window.signals.advanceLoadScreen.add(vsm._advanceLoader, vsm);
+        window.signals.hideLoadScreen.add(vsm._hideLoader, vsm);
+
+        window.signals.showMessageBox.add(vsm._showMessageBox, vsm);
+        window.signals.holdHideMessageBox.add(vsm._holdHideMessageBox, vsm);
+        window.signals.hideMessageBox.add(vsm._forceHideMessageBox, vsm);
+
+        window.signals.stateLoadComplete.add(_onStateLoadComplete);
+        window.signals.requestToSwitchView.add(_switchViewTo);
+        window.signals.loadLevel.add(_setNextGameMap);
+
+        pt.input.gamePad = new lib.input.GamePad();
+        pt.input.gamePad.reset(854, 480);
+
+        pt.ugc.ugcManager = new pt.ugc.UGCManager();
+        pt.ugc.ugcHistory = new pt.ugc.UGCHistory();
+
+        pt.render.layerManager = new lib.display.LayerManager();
+
+        var bufferAttr = {
+            antialias: false,
+            stencil: false,
+            alpha: false,
+            graphicsMemory: true,
+            dirty: false
+        };
+
+        var tvBacking = new lib.display.Backing('tv', 1280, 720, bufferAttr);
+        var gpBacking = new lib.display.Backing('gamepad', 854, 480, bufferAttr);
+
+        pt.render.layerManager.addBacking(tvBacking, true, 0, 480);
+        pt.render.layerManager.addBacking(gpBacking, true, 0, 0);
+
+        pt.asset.messageStudioData = new lib.locale.MessageStudio('messageData/playtime.json');
+        var location = lib.locale.LocaleManager.getSystemLocale().split('_');
+
+        lib.layout.LayoutData.setLocale(location[0], location[1]);
+        pt.asset.messageStudioData.setLocale(location[0], location[1]);
+
+        pt.asset.initCaches();
+
+        window.signals.loadTransitionComplete.add(vsm.changeView, vsm);
+
+        pt.savedata.saveStateInit();
+        pt.savedata.saveDataInit();
+
+        _setLoadState(LoadState.UI);
+
+        _updateLoadState();
+    }
+    vsm.init = init;
+
+    function changeView(direction) {
+        if (direction && direction === ui.Fade.Out) {
+            enableHBM();
+            return;
+        }
+
+        if (_viewHistory.length > _maxHistoryLength) {
+            _viewHistory = _viewHistory.slice(-_maxHistoryLength);
+            _nextViewIndex = _viewHistory.length - 1;
+            _currentViewIndex = _nextViewIndex - 1;
+
+            console.log('[VSM.changeView] Trimming history:');
+            console.log(_viewHistory);
+        }
+
+        if (_nextViewIndex != _currentViewIndex && _nextViewIndex !== -1) {
+            _exitCurrentView();
+        } else {
+            console.log('[VSM.changeView] There is no next view');
+        }
+    }
+    vsm.changeView = changeView;
+
+    function getViewName(indexFromCurrent) {
+        if (typeof indexFromCurrent === "undefined") { indexFromCurrent = 0; }
+        return _viewHistory[_currentViewIndex + indexFromCurrent];
+    }
+    vsm.getViewName = getViewName;
+
+    function isDialogTransitioning() {
+        return _uiHandler.isTransitioning();
+    }
+    vsm.isDialogTransitioning = isDialogTransitioning;
+
+    function disableHBM() {
+        nwf.system.WiiUSystem.getInstance().homeButtonEnabled = false;
+    }
+    vsm.disableHBM = disableHBM;
+
+    function enableHBM() {
+        nwf.system.WiiUSystem.getInstance().homeButtonEnabled = !_blockHBM;
+    }
+    vsm.enableHBM = enableHBM;
+
+    function enableScreenDimming() {
+        window.nwf.display.DisplayManager.getInstance().getTVDisplay().dimmingEnabled = true;
+        window.nwf.display.DisplayManager.getInstance().getGamePadDisplay().dimmingEnabled = true;
+    }
+    vsm.enableScreenDimming = enableScreenDimming;
+
+    function disableScreenDimming() {
+        window.nwf.display.DisplayManager.getInstance().getTVDisplay().dimmingEnabled = false;
+        window.nwf.display.DisplayManager.getInstance().getGamePadDisplay().dimmingEnabled = false;
+    }
+    vsm.disableScreenDimming = disableScreenDimming;
+
+    function _showMessageBox(message, msgtype, dialogType, callback, buttonSound, dialogScreen) {
+        if (typeof msgtype === "undefined") { msgtype = ui.Dialogs.GenericDialogButton.GDB_NONE; }
+        if (typeof dialogType === "undefined") { dialogType = ui.Dialogs.GenericDialogType.GDT_ANIMATED_SMALL; }
+        if (typeof callback === "undefined") { callback = null; }
+        if (typeof buttonSound === "undefined") { buttonSound = null; }
+        if (typeof dialogScreen === "undefined") { dialogScreen = ui.Dialogs.GenericDialogScreen.GDS_BOTH; }
+        if (_uiHandler.active) {
+        }
+        _uiHandler.showMessageBox(message, msgtype, dialogType, callback, buttonSound, dialogScreen);
+
+        if (_currentView) {
+            _currentView.stage.uiHandler.block();
+        }
+    }
+    vsm._showMessageBox = _showMessageBox;
+    function _holdHideMessageBox(message) {
+        _uiHandler.holdHideMessageBox(message, _unblockOtherDialogs);
+    }
+    vsm._holdHideMessageBox = _holdHideMessageBox;
+    function _forceHideMessageBox() {
+        _uiHandler.hideMessageBox(_unblockOtherDialogs);
+    }
+    vsm._forceHideMessageBox = _forceHideMessageBox;
+
+    function _unblockOtherDialogs() {
+        if (_currentView) {
+            _currentView.stage.uiHandler.unblock();
+        }
+    }
+
+    function _exitCurrentView() {
+        if (_nextViewIndex > 0) {
+            _currentView.exit();
+
+            reportStateDuration();
+        }
+
+        var i, shared = false;
+        if (!vsm.forceClearDynamicData && _currentView && _currentView.sharedViews) {
+            i = _currentView.sharedViews.length;
+            while (i--) {
+                if (_currentView.sharedViews[i] === _viewHistory[_nextViewIndex])
+                    shared = true;
+            }
+        }
+
+        pt.render.layerManager.clearAllLayers(shared);
+
+        vsm.errMan.clearProxies();
+
+        if (!shared) {
+            pt.asset.clearDynamicCaches();
+            lib.layout.clearDynamicCaches();
+            lib.effect.clearDynamicCaches();
+            console.log('[VSM.changeView] Clear Dynamic Data');
+        }
+
+        _currentView = null;
+        _stateChangeBurnFrames = GC_BURN_FRAMES;
+
+        if (shared) {
+            lib.forceGC(lib.GCType.JS_SWEEP, true);
+        } else {
+            lib.forceGC(lib.GCType.FULL, true);
+        }
+
+        vsm.playReport.sync();
+
+        vsm.forceClearDynamicData = false;
+    }
+
+    function _enterCurrentView() {
+        console.log('[VSM.changeView] Loading ' + _viewHistory[_nextViewIndex]);
+
+        _currentViewIndex = _nextViewIndex;
+        _currentView = _viewInstances[_viewHistory[_currentViewIndex]];
+
+        pt.render.layerManager.addLayers(pt.render.renderDefLoader.getLayerConfigs(_currentView.data.layers) || []);
+
+        _currentView.enter();
+    }
+
+    function _updateBlocking() {
+        _loadScreen.draw();
+
+        if (_uiHandler.active) {
+            _uiHandler.update();
+            _uiHandler.draw();
+        }
+
+        _soundPlayer.update();
+
+        if (_loadScreen.blocking) {
+            window.webkitRequestAnimationFrame(_updateBlocking);
+        } else {
+            window.webkitRequestAnimationFrame(_update);
+        }
+    }
+
+    function _updateIdle() {
+        if (_loadScreen.active)
+            _loadScreen.draw();
+
+        if (--_stateChangeBurnFrames < 0) {
+            var allocStats = window.nwf.system.Stats.getMemoryAllocSizes();
+            if (allocStats.wkDefaultMaxAllocSize < vsm.LowMemThreshold.MIN_ALLOC_DEFAULT) {
+                lib.forceLog('[VSM] Error! LowMemThreshold.MIN_ALLOC_DEFAULT ( ' + allocStats.wkDefaultMaxAllocSize + ' )');
+                vsm.errMan.displayFatalError(nwf.system.SystemErrorCode.WEBKIT_MEM_ALLOC_FAIL);
+            }
+
+            _enterCurrentView();
+
+            window.webkitRequestAnimationFrame(_update);
+            return;
+        }
+
+        if (_uiHandler.active) {
+            _uiHandler.update();
+            _uiHandler.draw();
+        }
+
+        window.webkitRequestAnimationFrame(_updateIdle);
+    }
+
+    function _update() {
+        var chromiumPortFps = (window.ChromiumPort && window.ChromiumPort.getFrameRate) ? window.ChromiumPort.getFrameRate() : 60;
+        var chromiumPortFrame = _chromiumPortFrameCounter++;
+        var chromiumPortSimulationTicks = chromiumPortFps === 30 ? 2 : (chromiumPortFps === 120 ? ((chromiumPortFrame % 2 === 0) ? 1 : 0) : 1);
+        var chromiumPortTick;
+
+        if (!_currentView) {
+            _updateIdle();
+            return;
+        }
+
+        if (_loadScreen.blocking) {
+            _updateBlocking();
+            return;
+        }
+
+        var stage = _currentView.stage;
+
+        for (chromiumPortTick = 0; chromiumPortTick < chromiumPortSimulationTicks; ++chromiumPortTick) {
+            pt.savedata.saveStateUpdate();
+
+            if (++vsm.LowMemThreshold._timer > vsm.LowMemThreshold.INTERVAL) {
+                var memStats = nwf.system.Stats.getMemoryStats(false);
+                if (memStats.wkDefaultFreeMemory < vsm.LowMemThreshold.WARNING_DEFAULT) {
+                    if (memStats.wkDefaultFreeMemory < vsm.LowMemThreshold.FATAL_DEFAULT) {
+                        lib.forceLog('[VSM] Error! LowMemThreshold.FATAL_DEFAULT ( ' + memStats.wkDefaultFreeMemory + ' )');
+                        vsm.errMan.displayFatalError(nwf.system.SystemErrorCode.WEBKIT_MEM_ALLOC_FAIL);
+                    } else {
+                        lib.forceGC(lib.GCType.JS_SWEEP, false);
+                        vsm.LowMemThreshold._timer = vsm.LowMemThreshold.INTERVAL - 5;
+                    }
+                } else if (memStats.jscDefaultFreeMemory < vsm.LowMemThreshold.WARNING_JSC_OBJECT) {
+                    lib.forceGC(lib.GCType.JS_SWEEP, false);
+                    vsm.LowMemThreshold._timer = vsm.LowMemThreshold.INTERVAL - 15;
+                } else {
+                    vsm.LowMemThreshold._timer = 0;
+                }
+            }
+
+            if (!vsm.freeze) {
+                vsm.errMan.update();
+
+                if (!window.blockInput) {
+                    pt.input.gamePad.update();
+                } else {
+                    pt.input.gamePad.setToIdleState();
+                }
+
+                pt.ugc.ugcManager.update();
+                if (_ugcLaunchSequence) {
+                    _ugcLaunchSequence.update();
+                    if (!_ugcLaunchSequence.running) {
+                        _ugcLaunchSequence.destroy();
+                        _ugcLaunchSequence = null;
+                    }
+                }
+            }
+
+            pt.render.layerManager.update();
+
+            if (_currentView.active && !vsm.freeze) {
+                stage.update();
+            }
+
+            _soundPlayer.update();
+
+            if (_uiHandler.active) {
+                _uiHandler.update();
+            }
+        }
+
+        if (_currentView.active) {
+            stage.draw();
+        }
+
+        if (_loadScreen.active)
+            _loadScreen.draw();
+
+        if (_uiHandler.active) {
+            _uiHandler.draw();
+        }
+
+        window.webkitRequestAnimationFrame(_update);
+    }
+
+
+    function _switchViewTo(newView) {
+        if (typeof newView === "undefined") { newView = ''; }
+        var musicDef, mapAssets, themeDef;
+        var musicChannels = new Array(lib.sound.NUM_MUSIC_CHANNELS);
+        var channelIndex;
+
+        _blockHBM = true;
+        disableHBM();
+
+        vsm.errMan.setShowCommunicationErrors(false);
+        vsm.errMan.setShowBatteryWarning(false);
+
+        console.log('-------------------------------------------------------');
+
+        if (newView === '') {
+            if (_currentViewIndex === 0) {
+                console.warn('[VSM._switchViewTo] There is no view to go back to');
+                return;
+            }
+
+            newView = _viewHistory[_currentViewIndex - 1];
+            console.log('[VSM._switchViewTo] Moving to previous view: ' + newView);
+        } else if (newView === _viewHistory[_currentViewIndex]) {
+            console.warn('[VSM._switchViewTo] You are already in ' + newView + ', but continuing anyway');
+        }
+
+        _nextViewIndex = _viewHistory.length;
+        _viewHistory.push(newView);
+
+        if (_nextViewIndex > _currentViewIndex + 1) {
+            console.warn('[VSM._switchViewTo] Skipping ahead to ' + _viewHistory.pop() + '?! Stopped');
+            _nextViewIndex = _currentViewIndex;
+            return;
+        }
+
+        pt.input.gamePad.reset();
+
+        _loadScreen.setFadeDuration(ui.FadeDuration.NORMAL);
+
+        console.log('[VSM._switchViewTo] Next view at ' + _nextViewIndex + ' (currently at ' + _currentViewIndex + ')');
+        if (_currentViewIndex === -1) {
+            changeView();
+
+            _update();
+        } else {
+            switch (newView) {
+                case 'game':
+                    if (_viewHistory[_currentViewIndex] === 'cutscene') {
+                        mapAssets = pt.asset.assetDefLoader.getMapAssets();
+                        themeDef = mapAssets.getThemeTypeDef(pt.currMapDef.settings.theme);
+                        musicDef = mapAssets.getThemeMusicDef(themeDef, pt.currMapDef.settings.music);
+
+                        if (!lib.sound.SoundPlayer.instance.isMusicPlaying(musicDef.intro)) {
+                            lib.sound.SoundPlayer.instance.playMusic(musicDef.intro, true);
+                        }
+                    }
+                    if (_viewHistory[_currentViewIndex] === 'editor') {
+                        _loadScreen.setState(ui.LoadScreenState.Black, ui.ProgressType.Saving);
+                        _loadScreen.setFadeDuration(ui.FadeDuration.FAST);
+                    } else {
+                        if (pt.currGameMode === pt.GAME_MODE_SOLO || pt.currGameMode === pt.GAME_MODE_SOLO_NEXT) {
+                            _loadScreen.setState(ui.LoadScreenState.Solo, ui.ProgressType.Loading);
+                        } else {
+                            _loadScreen.setState(ui.LoadScreenState.ProTip, ui.ProgressType.Loading, pt.ugc.mode === pt.ugc.Mode.Workshop ? ui.ProTipType.Editor : ui.ProTipType.Community);
+                        }
+                    }
+
+                    break;
+                case 'editor':
+                    if (_viewHistory[_currentViewIndex] === 'game' && !vsm.forceClearDynamicData) {
+                        _loadScreen.setState(ui.LoadScreenState.Black, ui.ProgressType.Saving);
+                        _loadScreen.setFadeDuration(ui.FadeDuration.FAST);
+                    } else {
+                        _loadScreen.setState(ui.LoadScreenState.ProTip, ui.ProgressType.Loading, ui.ProTipType.Editor);
+                    }
+                    break;
+                case 'level_select':
+                    if (_viewHistory[_currentViewIndex] === 'cutscene' && pt.sceneKey !== 'credits') {
+                        lib.sound.SoundPlayer.instance.stopAllMusic(5);
+
+                        lib.sound.SoundPlayer.instance.playBgmSelect('A005_LEVEL_SELECT_W1');
+                    }
+                    _loadScreen.setState(ui.LoadScreenState.ProTip, ui.ProgressType.Loading, ui.ProTipType.Solo);
+                    break;
+                case 'bonus_select':
+                    _loadScreen.setState(ui.LoadScreenState.ProTip, ui.ProgressType.Loading, ui.ProTipType.Solo);
+                    break;
+
+                case 'fishbowl':
+                    _loadScreen.setState(ui.LoadScreenState.ProTip, ui.ProgressType.Loading, pt.ugc.mode === pt.ugc.Mode.Workshop ? ui.ProTipType.Editor : ui.ProTipType.Community);
+                    break;
+                case 'cutscene':
+                    _loadScreen.setState(ui.LoadScreenState.Black, ui.ProgressType.None);
+                    _loadScreen.setFadeDuration(ui.FadeDuration.FAST);
+                    break;
+
+                default:
+                    _loadScreen.setState(ui.LoadScreenState.Normal, ui.ProgressType.Loading);
+                    break;
+            }
+
+            _loadScreen.fadeIn();
+        }
+    }
+
+    function _setLoadState(loadState) {
+        if (_loadState !== LoadState.GC) {
+            _loadStateQue = loadState;
+            _loadState = LoadState.GC;
+        } else {
+            _loadState = loadState;
+        }
+
+        switch (_loadState) {
+            case LoadState.UI:
+                lib.display.ShaderManager.loadShaders(lib.util.CanvasUtil.ctx);
+
+                _loadScreen = new ui.LoadingScreen();
+
+                _uiHandler = new ui.GenericDialogHandler();
+
+                pt.asset.preloadAllLayoutData();
+
+                break;
+            case LoadState.Init:
+                pt.storage.init();
+
+                vsm.playReport.init();
+
+                break;
+            case LoadState.Sound:
+                _soundPlayer = lib.sound.SoundPlayer.instance = new lib.sound.SoundPlayer('audio/json/', 'audio/sounds/');
+
+                break;
+            case LoadState.Actors_1:
+                pt.asset.preloadAllEntityActorData(pt.asset.ActorLoadPhase.First);
+
+                break;
+            case LoadState.Actors_2:
+                pt.asset.preloadAllEntityActorData(pt.asset.ActorLoadPhase.Last);
+
+                break;
+            case LoadState.RenderData:
+                pt.render.renderDefLoader = new pt.render.RenderDefLoader();
+                pt.render.renderDefLoader.load('json/layerdef.json');
+
+                break;
+            case LoadState.ViewData:
+                lib.curl(['json!json/views.json'], _onViewDataLoaded);
+
+                break;
+            case LoadState.AssetDef:
+                pt.asset.assetDefLoader = new pt.map.AssetDefLoader();
+                pt.asset.assetDefLoader.load('json/map/entitydef.json', 'json/map/themedef.json', 'json/map/tiledef.json', 'json/map/musicdef.json', 'json/map/PlaytimeLevels.json');
+
+                break;
+            case LoadState.Themes:
+                pt.asset.preloadGlobalTileData();
+
+                break;
+            case LoadState.Music:
+                _musicLoadParams = null;
+
+                break;
+            case LoadState.AllCaches:
+                break;
+            case LoadState.UGCAskPermission:
+                _permissionsFlow = new pt.ugc.UGCPermissionFlow();
+                _permissionsFlow.ugcAskPermission();
+                _loadScreen['_hideSysAnims']();
+                window.blockInput = false;
+                vsm.errMan.setShowCommunicationErrors(true);
+                break;
+            case LoadState.UGCExplainRestriction:
+                _loadScreen['_hideSysAnims']();
+                window.blockInput = false;
+                _explainRestriction();
+
+                break;
+            case LoadState.Done:
+                vsm.playReport.addLifetimeStat(PT1.runs, 1);
+                vsm.playReport.setSessionStat(PT1.region, nwf.system.WiiUSystem.getInstance().regionCode);
+                vsm.playReport.setSessionStat(PT1.lang, nwf.system.WiiUSystem.getInstance().languageCode);
+                vsm.playReport.setSessionStat(PT1.cnt, nwf.system.WiiUSystem.getInstance().countryCode);
+                vsm.playReport.setSessionStat(PT1.reload, nwf.system.WiiUSystem.getInstance().reloadCount);
+                vsm.playReport.addLifetimeStat(PT1.reload, nwf.system.WiiUSystem.getInstance().reloadCount > 0 ? 1 : 0);
+
+                if (!vsm.connected)
+                    vsm.playReport.addLifetimeStat(PT1.offline_net, 1);
+                if (vsm.miiverseBlocked)
+                    vsm.playReport.addLifetimeStat(PT1.offline_pc, 1);
+
+                if (pt.ugc.ugcManager.isLaunchedFromMiiverse() && pt.storage.getUGCCurrent().permissionGranted && nwf.act.NintendoAccountManager.getInstance().getActiveAccount().networkCommunicationAllowed) {
+                    pt.ugc.mode = pt.ugc.Mode.Community;
+                    ui.ugc.fromTitle = false;
+                    ui.ugc.fromBoot = true;
+                    _switchViewTo('fishbowl');
+                } else {
+                    _switchViewTo(_viewData.startView);
+                }
+
+                break;
+            case LoadState.GC:
+                if (_loadStateQue === LoadState.Init) {
+                    _loadScreen.setState(ui.LoadScreenState.ProTip, ui.ProgressType.FourSteps, ui.ProTipType.Solo);
+                    _loadScreen.fadeIn();
+                    _loadStateBurnFrames = GC_FADE_BURN_FRAMES;
+                } else if (_loadStateQue === LoadState.Actors_2) {
+                    window.signals.advanceLoadScreen.dispatch();
+                    _loadScreen.changeTip(ui.ProTipType.Community);
+                    _loadStateBurnFrames = GC_FADE_BURN_FRAMES;
+
+                    lib.util.dom.removeElement('#gp-container');
+                    lib.util.dom.removeElement('#tv-container');
+                    lib.util.dom.removeElement('#chromium-boot-overlay');
+                    lib.util.dom.removeElement('#loadscreen-style');
+                } else if (_loadStateQue === LoadState.RenderData) {
+                    if (!pt.ugc.ugcManager.isLaunchedFromMiiverse()) {
+                        window.signals.advanceLoadScreen.dispatch();
+                        _loadScreen.changeTip(ui.ProTipType.Editor);
+                    }
+                    _loadStateBurnFrames = GC_FADE_BURN_FRAMES;
+                } else if (_loadStateQue === LoadState.AllCaches) {
+                    window.signals.advanceLoadScreen.dispatch();
+                    if (pt.ugc.ugcManager.isLaunchedFromMiiverse()) {
+                        _loadScreen.changeTip(ui.ProTipType.Editor);
+                    }
+                    _loadStateBurnFrames = GC_FADE_BURN_FRAMES;
+                } else {
+                    _loadStateBurnFrames = GC_BURN_FRAMES;
+                }
+
+                if (_loadStateQue === LoadState.AllCaches) {
+                    lib.forceGC(lib.GCType.FULL, false);
+                } else {
+                    lib.forceGC(lib.GCType.FULL_NO_CODE, false);
+                }
+
+                break;
+        }
+
+        console.log('[VSM] Set Load State >> ' + LoadState[_loadState]);
+    }
+
+    function _updateLoadState() {
+        switch (_loadState) {
+            case LoadState.UI:
+                if (_loadScreen.ready && lib.layout.layoutDataSetCache.isReady() && _uiHandler.isReady()) {
+                    _setLoadState(LoadState.Init);
+                }
+
+                break;
+            case LoadState.Init:
+                if (pt.storage.isReady) {
+                    if (pt.storage.isCorrupt && vsm.errMan) {
+                        vsm.errMan.displayAlert(pt.asset.messageStudioData.getText('savememory_main_corrupt').text, pt.asset.messageStudioData.getText('btn_OK_01').text);
+                    }
+
+                    var ugcCurrent = pt.storage.getUGCCurrent();
+                    var activeAccount = nwf.act.NintendoAccountManager.getInstance().getActiveAccount();
+
+                    var postCache = pt.ugc.ugcManager.getPostCache(pt.ugc.PostCacheType.Shared);
+                    for (var levelIndex = 0; levelIndex < pt.ugc.MAX_MYLEVEL_COUNT; ++levelIndex) {
+                        var workshopStatus = pt.storage.getWorkshopStatus(levelIndex);
+                        if (workshopStatus.shared) {
+                            pt.ugc.ugcHistory.addLevelReference(workshopStatus.postID);
+                            pt.ugc.ugcHistory.setLevelYeahCount(workshopStatus.postID, workshopStatus.yeahs);
+                            pt.ugc.ugcHistory.setLevelReplyCount(workshopStatus.postID, workshopStatus.replyCount);
+
+                            var postItem = postCache.getItem(levelIndex);
+                            postItem.setSavedPostID(workshopStatus.postID);
+                            postItem.setSavedPosterID(activeAccount.principalID);
+                        }
+                    }
+
+                    postCache = pt.ugc.ugcManager.getPostCache(pt.ugc.PostCacheType.Saved);
+                    for (var levelIndex = 0; levelIndex < pt.ugc.MAX_SAVE_COUNT; ++levelIndex) {
+                        var ugcProgress = pt.storage.getUGCProgress(levelIndex);
+                        if (ugcProgress.postID) {
+                            pt.ugc.ugcHistory.addLevelReference(ugcProgress.postID);
+                            pt.ugc.ugcHistory.markLevelYeahed(ugcProgress.postID, ugcProgress.yeahed);
+                            pt.ugc.ugcHistory.setLevelYeahCount(ugcProgress.postID, ugcProgress.yeahs);
+                            pt.ugc.ugcHistory.markLevelPlayed(ugcProgress.postID, ugcProgress.played);
+                            pt.ugc.ugcHistory.setLevelTrophy(ugcProgress.postID, ugcProgress.trophy);
+                            pt.ugc.ugcHistory.setLevelReplyCount(ugcProgress.postID, ugcProgress.replyCount);
+
+                            var postItem = postCache.getItem(levelIndex);
+                            postItem.setSavedPostID(ugcProgress.postID);
+                            postItem.setSavedPosterID(ugcProgress.posterID);
+                        }
+                    }
+
+                    if (vsm.connected) {
+                        if (ugcCurrent.permissionGranted) {
+                            _ugcLaunchSequence = new pt.ugc.LaunchSequence();
+                            _ugcLaunchSequence.run();
+                        } else {
+                            if (pt.ugc.ugcManager.isLaunchedFromMiiverse()) {
+                                pt.ugc.ugcManager.trySetupMiiverse();
+                            }
+                        }
+                    }
+
+                    _setLoadState(LoadState.Sound);
+                }
+
+                break;
+            case LoadState.Sound:
+                if (_soundPlayer.areJSONLoaded && !_loadingMusicPlaying) {
+                    _loadingMusicPlaying = true;
+                    _soundPlayer.playMusic('LOADING', true, 1.0, true, true);
+                } else if (_soundPlayer.isLoaded()) {
+                    _setLoadState(LoadState.Actors_1);
+                }
+
+                break;
+            case LoadState.Actors_1:
+                if (pt.asset.actorDataCache.isReady() && pt.asset.patternActorDataCache.isReady()) {
+                    _setLoadState(LoadState.Actors_2);
+                }
+                break;
+            case LoadState.Actors_2:
+                if (pt.asset.actorDataCache.isReady() && pt.asset.patternActorDataCache.isReady()) {
+                    _setLoadState(LoadState.RenderData);
+                }
+
+                break;
+            case LoadState.RenderData:
+                if (pt.render.renderDefLoader.areRenderDefFilesLoaded()) {
+                    _setLoadState(LoadState.ViewData);
+                }
+
+                break;
+            case LoadState.ViewData:
+                break;
+            case LoadState.AssetDef:
+                if (pt.asset.assetDefLoader.areAssetDefFilesLoaded()) {
+                    _setLoadState(LoadState.Themes);
+                }
+
+                break;
+            case LoadState.Themes:
+                if (pt.asset.sharedTileDataCache.isReady()) {
+                    _setLoadState(LoadState.Music);
+                }
+
+                break;
+            case LoadState.Music:
+                if (_musicLoadParams === null || _musicLoadParams.isLoaded) {
+                    _setLoadState(LoadState.AllCaches);
+                }
+
+                break;
+            case LoadState.AllCaches:
+                if (pt.asset.areCachesReady()) {
+                    if (pt.ugc.ugcManager.isLaunchedFromMiiverse()) {
+                        if (!pt.storage.getUGCCurrent().permissionGranted) {
+                            _setLoadState(LoadState.UGCAskPermission);
+                        } else if (!nwf.act.NintendoAccountManager.getInstance().getActiveAccount().networkCommunicationAllowed) {
+                            _setLoadState(LoadState.UGCExplainRestriction);
+                        } else {
+                            _setLoadState(LoadState.Done);
+                        }
+                    } else {
+                        _setLoadState(LoadState.Done);
+                    }
+                }
+
+                break;
+            case LoadState.UGCAskPermission:
+                if (_permissionsFlow) {
+                    vsm.errMan.update();
+                    _permissionsFlow.update();
+
+                    if (_permissionsFlow.permissionsAccepted || _permissionsFlow.permissionsCanceled) {
+                        window.blockInput = true;
+                        vsm.errMan.setShowCommunicationErrors(false);
+                        if (_permissionsFlow.permissionsCanceled)
+                            window.signals.advanceLoadScreen.dispatch();
+                        _loadScreen['_showSysAnims']();
+                        _permissionsFlow.destroy();
+                        _permissionsFlow = null;
+                        _setLoadState(LoadState.Done);
+                    }
+                }
+
+                break;
+            case LoadState.GC:
+                if (--_loadStateBurnFrames < 0) {
+                    _setLoadState(_loadStateQue || 0);
+                }
+
+                break;
+        }
+
+        if (_loadScreen && _loadScreen.active) {
+            _loadScreen.draw();
+        }
+
+        if (_uiHandler && _uiHandler.active) {
+            if (!window.blockInput) {
+                pt.input.gamePad.update();
+            } else {
+                pt.input.gamePad.setToIdleState();
+            }
+            _uiHandler.update();
+            _uiHandler.draw();
+        }
+
+        pt.ugc.ugcManager.update();
+        if (_ugcLaunchSequence) {
+            _ugcLaunchSequence.update();
+            if (!_ugcLaunchSequence.running) {
+                _ugcLaunchSequence.destroy();
+                _ugcLaunchSequence = null;
+            }
+        }
+
+        if (_soundPlayer) {
+            _soundPlayer.update();
+        }
+
+        if (_loadState != LoadState.Done) {
+            window.webkitRequestAnimationFrame(_updateLoadState);
+        }
+    }
+
+    function _onViewDataLoaded(localViewData) {
+        if (!localViewData)
+            throw new Error('[VSM._onViewDataLoaded] views.json did not load!');
+
+        _viewData = localViewData;
+        _views = _viewData.views;
+
+        var viewArray = [];
+        var namespaceName;
+        var className;
+
+        for (var v in _views) {
+            namespaceName = _views[v].run['namespace'];
+            className = _views[v].run['class'];
+            if (window[namespaceName] && window[namespaceName][className]) {
+                _viewInstances[v] = new vsm.View(v, _views[v]);
+                viewArray.push(v);
+            }
+        }
+
+        viewArray = null;
+
+        _setLoadState(LoadState.AssetDef);
+    }
+
+    function _onStateLoadComplete() {
+        if (_loadScreen.active) {
+            console.log('[VSM._onStateLoadComplete] Fading out load screen...');
+            _loadScreen.fadeOut();
+        } else {
+            window.blockInput = false;
+        }
+
+        if (!vsm.forceHBMDisabled) {
+            _blockHBM = false;
+        }
+        vsm.forceHBMDisabled = false;
+
+        if (_currentView) {
+            if (_currentView.data.communication) {
+                vsm.errMan.setShowCommunicationErrors(true);
+            }
+
+            vsm.errMan.setShowBatteryWarning(true);
+        }
+    }
+
+    function _setNextGameMap(mapDef, gameMode, worldIndex, levelIndex) {
+        pt.currMapDef = mapDef;
+        pt.currWorldIndex = worldIndex;
+        pt.currLevelIndex = levelIndex;
+        pt.currGameMode = gameMode;
+        console.log('[VSM] Setting GameMode to ' + gameMode);
+
+        if (gameMode === pt.GAME_MODE_SOLO) {
+            if (pt.isCutsceneNeeded(worldIndex, levelIndex)) {
+                pt.sceneExitTo = 'game';
+            } else {
+                window.signals.requestToSwitchView.dispatch('game');
+                return;
+            }
+            window.signals.requestToSwitchView.dispatch('cutscene');
+        } else {
+            window.signals.requestToSwitchView.dispatch('game');
+        }
+    }
+    vsm._setNextGameMap = _setNextGameMap;
+
+    function _showLoader(state, progressType, proTipType, fadeDuration) {
+        if (typeof state === "undefined") { state = ui.LoadScreenState.Normal; }
+        if (typeof progressType === "undefined") { progressType = ui.ProgressType.Loading; }
+        if (typeof proTipType === "undefined") { proTipType = ui.ProTipType.None; }
+        if (typeof fadeDuration === "undefined") { fadeDuration = ui.FadeDuration.NORMAL; }
+        _loadScreen.setState(state, progressType, proTipType);
+        _loadScreen.setFadeDuration(fadeDuration);
+        _loadScreen.fadeIn();
+    }
+    vsm._showLoader = _showLoader;
+
+    function _advanceLoader() {
+        _loadScreen.goToNextMarker();
+    }
+    vsm._advanceLoader = _advanceLoader;
+
+    function _hideLoader() {
+        _loadScreen.fadeOut();
+    }
+    vsm._hideLoader = _hideLoader;
+
+    function _onSetupMiiverse(taskResult) {
+        _miiverseSetupComplete = taskResult;
+        pt.ugc.ugcManager.removeListener(pt.ugc.TaskType.SetupMiiverse, _onSetupMiiverse);
+    }
+
+    function _explainRestriction() {
+        var msg = pt.asset.messageStudioData.getText('restrict_online_interaction');
+        window.signals.showMessageBox.dispatch(msg, ui.Dialogs.GenericDialogButton.GDB_OK, ui.Dialogs.GenericDialogType.GDT_FULL, _onExplainRestrictionExit);
+    }
+
+    function _onExplainRestrictionExit() {
+        window.blockInput = true;
+        _loadScreen['_showSysAnims']();
+        _setLoadState(LoadState.Done);
+    }
+})(vsm || (vsm = {}));
+//@ sourceMappingURL=ViewStateManager.js.map
