diff --git a/scripts/ui/ugc/KeyboardController.js b/scripts/ui/ugc/KeyboardController.js
index 23319e9..5d3c450 100644
--- a/scripts/ui/ugc/KeyboardController.js
+++ b/scripts/ui/ugc/KeyboardController.js
@@ -35,26 +35,28 @@ var ui;
             this.y = deltaY;
         };
 
-        KeyboardController.prototype.onKeyDown = function (evt) {
-            if (window.blockInput)
-                return;
-
-            this.down = true;
-            switch (evt.keyCode) {
-                case 37:
-                    deltaX = speed;
-                    break;
-                case 39:
+        KeyboardController.prototype.onKeyDown = function (evt) {
+            if (window.blockInput)
+                return;
+
+            switch (evt.keyCode) {
+                case 37:
+                    deltaX = speed;
+                    break;
+                case 39:
                     deltaX = -speed;
                     break;
                 case 38:
                     deltaY = speed;
                     break;
-                case 40:
-                    deltaY = -speed;
-                    break;
-            }
-        };
+                case 40:
+                    deltaY = -speed;
+                    break;
+                default:
+                    return;
+            }
+            this.down = true;
+        };
 
         KeyboardController.prototype.onKeyUp = function (evt) {
             switch (evt.keyCode) {
