diff --git a/scripts/pt/GameState.js b/scripts/pt/GameState.js
index 56cdee0..181cd72 100644
--- a/scripts/pt/GameState.js
+++ b/scripts/pt/GameState.js
@@ -174,11 +174,11 @@ var pt;
             switch (this.substate) {
                 case SubState.AssetDefs:
                     break;
-                case SubState.Shared:
-                    if (pt.currGameMode === pt.GAME_MODE_EDIT_PLAYTEST) {
-                        backgroundName = 'EditPlaytest';
-                        playTestMenu = true;
-                    } else if (pt.currGameMode === pt.GAME_MODE_MY_LEVELS) {
+                case SubState.Shared:
+                    if (pt.currGameMode === pt.GAME_MODE_EDIT_PLAYTEST) {
+                        backgroundName = this.mapDef.settings.background || 'EditPlaytest';
+                        playTestMenu = true;
+                    } else if (pt.currGameMode === pt.GAME_MODE_MY_LEVELS) {
                         backgroundName = this.mapDef.settings.background;
                         playTestMenu = !pt.storage.getWorkshopStatus(pt.currLevelIndex).tested;
                     } else {
