diff --git a/scripts/lib/display/Layer.js b/scripts/lib/display/Layer.js
index 1909dc2..76667ec 100644
--- a/scripts/lib/display/Layer.js
+++ b/scripts/lib/display/Layer.js
@@ -24,8 +24,8 @@ var lib;
                 this.type = LayerType.DIRECT;
                 this.position = new lib.math.Vec2(0, 0);
                 this.shared = false;
-                this._dirty = false;
-                this._needsSorting = false;
+                this._dirty = false;
+                this._needsSorting = false;
                 this._topFrame = false;
                 this._width = 854;
                 this._height = 480;
@@ -329,13 +329,13 @@ var lib;
                     ctx.imageSmoothingEnabled = this.imageSmoothingEnabled;
                 }
 
-                if (this.type === LayerType.STATIC && !this._dirty) {
-                    this._renderLayer(this._buffer);
-                } else if (this.type === LayerType.BUFFER && this._topFrame) {
-                    if (this._dirty && this._autoClear)
-                        this._clearBuffer();
-
-                    this._renderLayer(this._buffer);
+                if (this.type === LayerType.STATIC && !this._dirty) {
+                    this._renderLayer(this._buffer);
+                } else if (this.type === LayerType.BUFFER && this._topFrame) {
+                    if (this._autoClear)
+                        this._clearBuffer();
+
+                    this._renderLayer(this._buffer);
                 }
 
                 if (this.camera) {
@@ -354,15 +354,21 @@ var lib;
                 this._dirty = false;
             };
 
-            Layer.prototype._renderLayer = function (ctx) {
-                var child = this._children.head;
-
-                while (child) {
-                    if (child.visible)
-                        child.render(ctx);
-
-                    child = child.next;
-                }
+            Layer.prototype._renderLayer = function (ctx) {
+                var child = this._children.head;
+
+                while (child) {
+                    if (child.visible) {
+                        ctx.save();
+                        try  {
+                            child.render(ctx);
+                        } finally {
+                            ctx.restore();
+                        }
+                    }
+
+                    child = child.next;
+                }
 
                 this._dirty = true;
                 this._topFrame = false;
