diff --git a/scripts/lib/layout/WindowPane.js b/scripts/lib/layout/WindowPane.js
index b94f663..7145334 100644
--- a/scripts/lib/layout/WindowPane.js
+++ b/scripts/lib/layout/WindowPane.js
@@ -670,8 +670,9 @@ var lib;
                 this._prerender = this._prerenderCTX.canvas;
                 this.preRender();
             };
-            WindowPane.prototype.preRender = function () {
-                this._prerenderCTX.clear();
+            WindowPane.prototype.preRender = function () {
+                this._prerenderCTX.canvas.width = this._prerenderCTX.canvas.width;
+                this._prerenderCTX.clear();
                 for (var i = 0, len = this._pieces.length; i < len; ++i) {
                     var piece = this._pieces[i];
                     if (piece._texture === null)
