diff --git a/scripts/ui/GameUI.js b/scripts/ui/GameUI.js
index 916e1b4..25e1285 100644
--- a/scripts/ui/GameUI.js
+++ b/scripts/ui/GameUI.js
@@ -51,7 +51,7 @@ var ui;
         GameUI.prototype.load = function () {
             this.hud.load('game_hud', 'game_hudTV');
             this.gameoverMenu.load('game_over', 'game_over_TV');
-            this.pauseMenu.load('My_pause', 'My_pause_TV');
+            this.pauseMenu.load('my_pause', 'my_pause_TV');
             this.resultMenu.load('result', 'result_TV');
 
             this._startState(GameUI.STATE_LOADING);
